tools/payments/client.go

```diff
@@ -0,0 +1,185 @@
+	submitConfigStream  = submitPrefix + "-configure-" + payout
+)
+
+// redisClient is an implementation of settlement client using clustered redis client
+type redisClient struct {
+	env   string
+	redis *redis.ClusterClient
+}
+
+func newRedisClient(ctx context.Context, env string) (*redisClient, error) {
+	tlsConfig := &tls.Config{
+		ServerName: "redis",
+		MinVersion: tls.VersionTLS12,
+		ClientAuth: 0,
+	}
+
+	addrs := strings.Split(os.Getenv("REDIS_ADDRS"), ",")
+	pass := os.Getenv("REDIS_PASS")
+	username := os.Getenv("REDIS_USERNAME")
+
+	// only if environment is local do we hardcode these values
+	if env == "local" {
+		certPool := x509.NewCertPool()
+		pem, err := ioutil.ReadFile("redistest/test/redis/tls/ca.crt")
+		if err != nil {
+			return nil, fmt.Errorf("failed to read test-mode ca.crt: %w", err)
+		}
+		certPool.AppendCertsFromPEM(pem)
+		tlsConfig.RootCAs = certPool
+	}
+
+	rc := &redisClient{
+		env: env,
+		redis: redis.NewClusterClient(
+			&redis.ClusterOptions{
+				Addrs: addrs, Password: pass, Username: username,
+				DialTimeout:     15 * time.Second,
+				WriteTimeout:    5 * time.Second,
+				MaxRetries:      5,
+				MinRetryBackoff: 5 * time.Millisecond,
+				MaxRetryBackoff: 500 * time.Millisecond,
+				PoolSize:        10,
+				PoolTimeout:     30 * time.Second,
+				TLSConfig:       tlsConfig,
+			}),
+	}
+	err := rc.redis.Ping(ctx).Err()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping redis: %w", err)
+	}
+	return rc, nil
+}
+
+// ConfigureWorker implements settlement client
+func (rc *redisClient) ConfigureWorker(ctx context.Context, stream string, config *WorkerConfig) error {
+	_, err := rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: stream,
+			Values: map[string]interface{}{
+				"data": config}},
+	).Result()
+	if err != nil {
+		return fmt.Errorf("failed to push config to workers: %w", err)
+	}
+	return nil
+}
+
+// PrepareTransaction implements settlement client
+func (rc *redisClient) PrepareTransaction(ctx context.Context, t *PrepareTx) error {
+	body, err := json.Marshal(t)
+	if err != nil {
+		return fmt.Errorf("failed to serialize transaction: %w", err)
+	}
+
+	// message wrapper for prepare
+	message := &prepareWrapper{
+		ID:        uuid.New(),
+		Timestamp: time.Now(),
+		Body:      string(body),
+	}
+
+	// add to stream
+	_, err = rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: prepareStream,
+			Values: map[string]interface{}{
+				"data": message}},
+	).Result()
+	if err != nil {
+		return fmt.Errorf("failed to prepare transaction: %w", err)
+	}
+	return nil
+}
+
+// SubmitTransaction implements settlement client
+func (rc *redisClient) SubmitTransaction(ctx context.Context, signer httpsignature.ParameterizedSignator, at *AttestedTx) error {
+	body, err := json.Marshal(at)
+	if err != nil {
+		return fmt.Errorf("failed to marshal attested transaction body: %w", err)
+	}
+	// create a request
+	req, err := http.NewRequest("POST", rc.env+"/authorize", bytes.NewBuffer(body))
```

Comment:
There is a slight difference between using the `http.NewRequest` and the `NewRequest` in the lib/clients package, the later uses 
```
buf := new(bytes.Buffer)
err := json.NewEncoder(buf).Encode(expected)
```
to do the encoding which is different to using `json.Marshal` so the Content-Length will be different, possibly because `json.NewEncoder` adds a new line or something. All of our clients use the lib client `NewRequest` so we may want to change `json.Marshal` to `json.NewEncoder`